pkgs/jupiter-hw-support/firmware.nix

```diff
@@ -103,6 +103,12 @@ stdenv.mkDerivation {
     wrapProgram $out/bin/jupiter-controller-update \
       --prefix PATH : ${lib.makeBinPath [ jq pythonEnv ]}
 
+    mkdir -p $out/libexec
+    makeShellWrapper \
+      $out/share/jupiter_controller_fw_updater/d20bootloader.py \
+      $out/libexec/d20bootloader \
+      --prefix PATH : ${lib.makeBinPath [ pythonEnv ]}
+
```

Comment:
This uses `libexec` to prevent accidental misuses. You have to willfully play with the libexec path to *do stuff*...

Though AFAICT there is nothing that should be breaking stuff... other than the serial number writing stuff. AFAICT the bootloader is never replaced, or erased[citation needed], and anyway it's a new script.